### PR TITLE
I was curious if extracting the logger creation made sense in this case

### PR DESCRIPTION
**Pull request by SvenDowideit**
Signed-off-by: Sven Dowideit <SvenDowideit@home.org.au>

Heya :)

I was talking to someone at a meetup last week, and I might be able to persuade them to write a very simplistic network based logger next month at the party - so I was looking at your PR, and wondered if it was worth extracting the logger creation code out of the container module.

I'm not really sure anymore, but as I need to learn more go-style, thought I'd ask - is this a reasonable and worthwhile extraction (on the assumption that there will be more logger drivers soon), or is that not how its done :)

### CODE REVIEW

File: daemon/logger/loggers/loggers.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package loggers
+
+import (
+	"fmt"
+	"github.com/docker/docker/daemon/logger"
+	"github.com/docker/docker/daemon/logger/dockerlog"
+)
+
+func New(Type string, Config map[string]string) (logger.Logger, error) {
```
Comment (LK4D4) [inline daemon/logger/loggers/loggers.go]: Parameters should start from lowercase.

**Comment by LK4D4** [comment]
I like it

**Comment by SvenDowideit** [comment]
yay :)

File: daemon/container.go [hunk 2]
```diff
@@ -1337,32 +1337,25 @@ func (container *Container) startLogging() error {
...
-		dl, err := dockerlog.New(pth)
-		if err != nil {
-			return err
-		}
-		l = dl
-	case "none":
-		return nil
-	default:
-		return fmt.Errorf("Unknown logging driver: %s", cfg.Type)
+	pth, err := container.logPath("json")
```
Comment (LK4D4) [inline daemon/container.go]: Now I think, so we're passing `config` for `default` even if type is `none`. How it will be if there will be more drivers?

Comment (SvenDowideit) [inline daemon/container.go]: mmm, true - I think I miss-interpreted `default` - I was kinda expecting that at some stage, an admin might be able to configure the daemon default to be `rsyslog` - and so the defaultConfig would become that endpoint info - and then one can over-ride that configurable default for individual containers.

LMK :)

Comment (LK4D4) [inline daemon/container.go]: No, "default" is name of driver :) `container.daemon.DefaultLogConfig` is default config. I really don't know what to do with `container.logPath` we can pass it to each driver anyway, but maybe call not "Path", but something more distinctive.

**Comment by SvenDowideit** [comment]
oh yay, lets try un-rebaseing that.

**Comment by SvenDowideit** [comment]
mmm, ugly. but is it better?

File: daemon/container.go [hunk 3]
```diff
@@ -1337,32 +1337,26 @@ func (container *Container) startLogging() error {
 	if cfg.Type == "" {
 		cfg = container.daemon.DefaultLogConfig
 	}
-	var l logger.Logger
-	switch cfg.Type {
-	case "default":
+	if cfg.Config == nil {
+		cfg.Config = make(map[string]string)
+	}
+	if cfg.Type == "" {
```
Comment (LK4D4) [inline daemon/container.go]: Hm, how it can be? I think that here at some point should be `default` or `none`.

Comment (SvenDowideit) [inline daemon/container.go]: sorry, I'm still jetlagged :( running tests....

**Comment by SvenDowideit** [comment]
I'm pretty puzzled - I can't get the tests to pass - both with just your changes and with mine added.

if I just run the `TestDaemonLogging` tests on either, they're both happy, but the full set error.

**Pull request closed without merging**